src/app/routes/applications/applications.component.ts

```diff
@@ -91,34 +94,37 @@ export class ApplicationsComponent implements OnInit, AfterViewInit, OnDestroy {
     this.cleanFilters();
   }
 
-  openNewOrgComponent(): void {
-    if (!this.isIamEwcOwner) {
-      const namespace =
-        'orgcreator.apps.testorg.' + this.envService.rootNamespace;
-      const roleName = 'org';
-      this.router.navigate([RouterConst.Enrol], {
-        queryParams: { roleName, app: namespace, stayLoggedIn: true },
-      });
-      return;
-    }
-
-    const dialogRef = this.dialog.open(NewOrganizationComponent, {
-      width: '600px',
-      data: {},
-      maxWidth: '100%',
-      disableClose: true,
-    });
-
-    dialogRef
-      .afterClosed()
-      .pipe(takeUntil(this.subscription$))
-      .subscribe((result) => {
-        // console.log('The dialog was closed');
+  handleNewOrgRequest(): void {
```

Comment:
```suggestion
  if (this.envService.production) {
    this.requestOrgCreationFromEmail()
   } else {
     if(!this.isIamEwcOwner) {
       this.requestOrgCreationFromEnrolment();
       return;
     } 
     this.createNewOrg();
   }
```
What do you think about split this method into few smaller?

Reply:
I split this into 2 methods